src/pages/Governance/hooks/useSubmitVote.ts

```diff
@@ -0,0 +1,64 @@
+import {Types} from "aptos";
+import {useState} from "react";
+
+export type VoteResponse = VoteResponseOnSuccess | VoteResponseOnFailure;
+
+export type VoteResponseOnSuccess = {
+  succeeded: boolean;
+  transactionHash: string;
+};
+
+export type VoteResponseOnFailure = {
+  succeeded: boolean;
+  message: string;
+};
+
+const processTransaction = async (
+  transactionPayload: Types.TransactionPayload,
+): Promise<VoteResponse> => {
+  const responseOnFailure = {
+    succeeded: false,
+    message: "Unknown Error",
+  };
+
+  try {
+    const response = await window.aptos.signAndSubmitTransaction(
+      transactionPayload,
```

Comment:
I feel this a wallet api and might be better to place it in the wallet api file...?

Reply:
Do you mean the `signAndSubmitTransaction` part should go to a wallet api file? Yeah agree. signing transactions is a rather general functionality and not just voting will need it. Let me add a TODO here. We can change it along with using BCS as you mentioned below.

Reply:
sounds good. I'll be working on the stake transaction so can implement those changes a part of that PR